### PR TITLE
Path refactoring

### PR DESCRIPTION
**Pull request by IrinaShkviro**
Used `JoinPath()` instead of separators

Added `std::filesystem::create_directory` instead of Path::MakeDir as was mentioned in TODO

### CODE REVIEW

File: OrbitCore/Path.cpp [hunk 1]
```diff
@@ -126,63 +117,63 @@ std::string Path::GetDllName(bool a_Is64Bit) {
   return a_Is64Bit ? "Orbit64.dll" : "Orbit32.dll";
 }
 
+static std::string GetConfigPath() {
```
Comment (beckerhe) [inline OrbitCore/Path.cpp:120]: Maybe something like `CreateAndGetConfigPath` to make it clear that this function does have side effects.

File: OrbitCore/Path.cpp [hunk 2]
```diff
@@ -252,7 +243,7 @@ std::string Path::GetProgramFilesPath() {
 #ifdef WIN32
```
Comment (dimitry-) [inline OrbitCore/Path.cpp:243]: This one also seems unused - only mentioned in dump.

Comment (IrinaShkviro) [reply OrbitCore/Path.cpp:243]: The full list of unused functions:

- GetOrbitAppPdb()
- GetLicenseName()
- GetTmpPath()
- GetProgramFilesPath()
- GetMainDrive()
- GetSourceRoot() - one more occurrence in PathTest.cpp
- IsSourceFile()

Should I remove them all?

Comment (beckerhe) [reply OrbitCore/Path.cpp:243]: Yes!

File: OrbitCore/Path.cpp (hunk 1 above)
Comment (IrinaShkviro) [reply OrbitCore/Path.cpp]: Renamed the function, thanks!

**Comment by IrinaShkviro** [comment]
> otherwise LGTM

Thanks for the review!

**Pull request merged**